packages/tags/src/views/Tag.js

```diff
@@ -5,110 +5,175 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
+import styled, { css } from 'styled-components';
+import { math, stripUnit } from 'polished';
 import PropTypes from 'prop-types';
-import styled from 'styled-components';
 import classNames from 'classnames';
-import TagStyles from '@zendeskgarden/css-tags';
-
-import { retrieveComponentStyles, isRtl } from '@zendeskgarden/react-theming';
+import { getColor, isRtl, retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import DEFAULT_THEME from '@zendeskgarden/react-theming/src/theme';
 
 const COMPONENT_ID = 'tags.tag_view';
 
-const TYPE = {
-  GREY: 'grey',
-  BLUE: 'blue',
-  KALE: 'kale',
-  RED: 'red',
-  GREEN: 'green',
-  YELLOW: 'yellow',
-  FUSCHIA: 'fuschia',
-  PINK: 'pink',
-  CRIMSON: 'crimson',
-  ORANGE: 'orange',
-  LEMON: 'lemon',
-  LIME: 'lime',
-  MINT: 'mint',
-  TEAL: 'teal',
-  AZURE: 'azure',
-  ROYAL: 'royal',
-  PURPLE: 'purple'
-};
-
 const SIZE = {
   SMALL: 'small',
+  MEDIUM: 'medium',
   LARGE: 'large'
 };
 
+const colorStyles = props => {
+  let backgroundColor;
+  let boxShadowColor;
+  let foregroundColor;
+
+  if (props.hue) {
+    const shade = props.hue === 'yellow' ? 400 : 600;
+
+    backgroundColor = getColor(props.hue, shade, props.theme);
+    boxShadowColor = getColor(props.hue, shade, props.theme, 0.35);
+
+    if (props.hue === 'yellow' || props.hue === 'lemon') {
+      foregroundColor = getColor('yellow', 800, props.theme);
+    } else {
+      foregroundColor = getColor('white');
+    }
+  } else {
+    backgroundColor = getColor('neutralHue', 200, props.theme);
+    boxShadowColor = getColor('neutralHue', 600, props.theme, 0.35);
+    foregroundColor = getColor('neutralHue', 700, props.theme);
+  }
+
+  return css`
+    background-color: ${backgroundColor};
+    color: ${foregroundColor};
+
+    &:hover {
+      color: ${foregroundColor}; /* <a> element reset */
+    }
+
+    &.focus-visible {
```

Comment:
The useFocusVisible hook prefixes the class with `garden` should we do that here too?

Reply:
Possibly. For now I want to leave these `.focus-visible` classes in place so that we can grep the codebase for locations (also in buttons) that should be addressed when `@zendeskgarden/container-focusvisible` is introduced. I'm also not exactly sure we want to retain the original `focused` assumptions that @austin94 made on this tags component – another topic to discuss as we add the `useFocusVisible` hook.